### DISCUSSION TITLE
Where are the range and value settings stored?

### DISCUSSION THREAD

**Discussion by mediahope** [rsjaffe/MIDI2LR#753] [Q&A] [answered]
Hey guys! Does anyone know where the settings of ranges and values for encoders are stored, which are configured via the PCM for each button? Just for some reason they are not in the xml configuration file. And there was a situation when everything was reset and had to be configured again.
![image](https://github.com/rsjaffe/MIDI2LR/assets/13403919/02cc55a9-0927-4e74-9d9d-446475e1ced0)

**Comment by rsjaffe** [comment, marked as answer]
The location differs based on whether you're using a Mac or a Windows PC. It is saved in the same directory as the MIDI2LR log file, so the easiest way to find it is to use the Lightroom menu **Help → Plug-in Extras** and select **Generate diagnostic report**. Look for *settings.xml*. That's the file.

**Reply by mediahope** [reply]
Thank you very much! I've been looking for this for a very long time!
